Fixes to build scripts for Windows.

AaronFriel opened the pull request:
Issue: On Windows platforms, the `dev` scripts in `.\app\react` and `vue` relied on three things that didn't work: 

1. They assumed the parent project's dependencies were installed in the `prepare` script. This is fixed by adding `prepare-root`.

2. The environment variable `DEV_BUILD` is set in a *nix-ish way, which doesn't work in PowerShell.

3. The `--exec` parameter of `nodemon` used single quotes instead of double quotes.

## What I did

I added a `prepare-root` script which builds the parent package, added that as an initial command to the `dev` script.

I added the `cross-env` dependency and used that to set the `DEV_BUILD` parameter.

I double quoted and escaped the `--exec` parameter.

I then tested these changes on a Linux environment. I used *nix style path separators in the `prepare-root` script.

AaronFriel committed:
Build scripts working on Windows.

ndelangen commented:
Would you care to review this @UsulPro ?

Hypnosphi commented on app/react/package.json:
```diff
@@ -18,7 +18,8 @@
   },
   "scripts": {
-    "dev": "DEV_BUILD=1 nodemon --watch ./src --exec 'yarn prepare'",
+    "prepare-root": "cd ../../ && yarn && cd ./app/react",
```
We use [yarn workspaces](https://yarnpkg.com/en/docs/workspaces), so you can run `yarn install` in any package directory, and it will bootstrap everything. The `cd` part is therefore redundant

AaronFriel replied:
Oh that's curious. It looks like Yarn just released 1.2.1, perhaps this was a bug fixed with workspaces.

AaronFriel replied:
Would you like me to remove the `prepare-root` script from this and retry from a clean git pull?

Hypnosphi replied:
Yes please, but you can actually do `yarn bootstrap --reset` instead of clean pull

AaronFriel replied:
Hey, I commented further below, could you check in on this?

ndelangen commented:
@AaronFriel Anything we need to do on our side to get this merged?

Who in the @storybooks/team works with windows and can do a review on this?
//cc @UsulPro 🙇

AaronFriel committed:
Run bootstrap script with node.

Hypnosphi commented:
There’re no such thing as “just work on @storybook/react package”. Even if it’s the only package where you make changes, you still need to run example app (located in `examples/cra-kitchen-sink`), eslint, and the tests. All of it requires bootstrapping (that is, installing the external dependencies, and cross-linking the internal ones)

AaronFriel commented:
Oh, I actually had significant success using the tool `yalc` to just work on one package, while working with another repository (react-native-web). Yalc works by symlinking in every file and folder from a source directory (e.g.: `storybook/react/app`) to the appropriate package directory. It allowed me to make changes to the react package while not having to work with the rest of the packages.

AaronFriel commented:
If it's a blocking issue for merging this, I can remove the `prepare-root` script. That said, it sounds like in your normal development workflow running yarn on the root directory is a no-op. It hurts nothing, and ensures `yarn` works in those directories.

Hypnosphi commented:
My point was that as though you CAN work with only one package being set up, you SHOULDN'T do that, because you have to test how this interacts with current state of other packages anyway.

So yes, please remove the `prepare-root` script

Hypnosphi commented:
By the way, maybe there's a sense in using `cross-env` in all the packages? We already have it in one: https://github.com/storybooks/storybook/blob/fad3ea3c907093ab62c56c7a7f6b9bb1e3367630/examples/vue-kitchen-sink/package.json

If you feel like that, you could add `cross-env` as a root dependency and prepend it to commands where needed. Feel free to contact us on slack if you have any questions about what some package does: https://now-examples-slackin-nqnzoygycp.now.sh/

AaronFriel committed:
Add cross-env as root devdep, remove unnecessary script.

AaronFriel commented:
I changed other uses of the pattern `[\w\d_]+=` in scripts, e.g. in `addons/comments/package.json`.

I added cross-env to the root project, and removed prepare-root scripts. I also made sure that the `vue-kitchen-sink`, which references `cross-env` in its scripts has it as a dev dependency.

Hypnosphi commented on examples/vue-kitchen-sink/package.json:
```diff
@@ -14,7 +14,7 @@
     "babel-preset-env": "^1.6.0",
     "babel-preset-vue": "^1.2.1",
-    "cross-env": "^3.0.0",
+    "cross-env": "^5.1.1",
```
Is it still needed here, given that we have it in root now?

AaronFriel replied:
I wasn't sure. The kitchen sink is something people should be able to play with in isolation, right?

I assumed that, because `babel-core` and `babel-preset-env` were also duplicated, that the package `vue-kitchen-sink` should be self-contained.

Hypnosphi replied:
>  The kitchen sink is something people should be able to play with in isolation, right?

Not really. In our case, it‘s mainly used to test the changes made to other packages

Hypnosphi commented:
Please run `yarn` in root to update `yarn.lock` file

AaronFriel commented:
I rebased on master.

AaronFriel committed:
Run test script with node.

AaronFriel committed:
Merge branch 'master' into winqol

AaronFriel committed:
Fix package.json in app/vue

AaronFriel committed:
Update yarn.lock.

AaronFriel commented:
Gosh that was fun. I think with that last commit all of the CI checks will be happy.

Hypnosphi commented:
Great, thanks @AaronFriel !

ndelangen commented:
Thank you @AaronFriel !
